docs/changelog/1712.misc.rst

```diff
@@ -0,0 +1 @@
+Do not pass around regex match object in Replacer. - by :user:`jayvdb`
```

Comment:
Only user-facing changes need changelog.

Reply:
great. But wont CI fail if this file doesnt exist?

Reply:
The check will fail but that's not mandatory for merge. You can add some naming of the title too to skip it I think 🤔

Reply:
Done